tests/phpunit/EngineTest.php

```diff
@@ -436,12 +426,10 @@ public function test_memory_limiter_is_applied_after_static_analysis_when_enable
 
     public function test_memory_limiter_is_not_applied_when_initial_tests_are_skipped(): void
     {
-        $config = $this->createMock(Configuration::class);
-        $config
-            ->expects($this->once())
-            ->method('shouldSkipInitialTests')
-            ->willReturn(true)
-        ;
+        $config = ConfigurationBuilder::withMinimalTestData()
+            ->withSkipInitialTests(true)
+            ->withWithUncovered(true)
+            ->build();
```

Comment:
We might have remove a tested assertion somewhere in this PR. Since we no longer make a full MT run, so there's a chance this updated test will miss some mutations that were otherwise caught.

Reply:
I think if anything the builder will _introduce_ assertions, the ones from `Configuration::__construct()`. The mock on the other hand does not have any

Reply:
The mock had `->expects($this->once())`.